discovery-provider/plugins/pedalboard/apps/relay/src/txRelay.ts

```diff
@@ -49,15 +48,33 @@ export const relayTransaction = async (
 
   log({ msg: "gathered tx params", nonce });
 
+  const gasLimit = 20000000000;
+
   // assemble, sign, and send transaction
   const transaction = { nonce, gasLimit, to, value, data };
   await senderWallet.signTransaction(transaction);
   const submit = await senderWallet.sendTransaction(transaction);
 
   log("signed and sent");
 
-  // internally polls until mined
-  const receipt = await submit.wait(requiredConfirmations);
-
+  // query chain until tx is mined
+  const receipt = await confirm(web3, submit.hash)
   return { receipt, transaction };
 };
+
+const confirm = async (web3: ethers.providers.JsonRpcProvider, txHash: string): Promise<TransactionReceipt> => {
+  const result = await Promise.race([confirmIndefinitely(web3, txHash), delay(6000)])
```

Comment:
what happens to the promise that doesn't finish first? is something still running it until it completes or promise.race kills it?

Reply:
wow great catch, race doesn't actually cancel the other promise

Reply:
damn they do not make that clear on the js docs lol